netket/optimizer/jax/stochastic_reconfiguration.py

```diff
@@ -62,9 +58,33 @@ def _jax_cg_solve(
     return out
 
 
-@jit
+@partial(jit, static_argnums=1)
+def _jax_cg_solve_onthefly(
+    x0,
+    forward_fn,
+    params,
+    samples,
+    grad,
+    diag_shift,
+    n_samp,
+    sparse_tol,
+    sparse_maxiter,
+):
```

Comment:
@PhilipVinc I could just add `n_samp` to `static_argnums` here (possibly also `diag_shift`, `sparse_tol`, `sparse_maxiter`)

Reply:
n_samp could be a static_argnum because it de-facto is. 
But, as I said in my comment above, it is a redundant parameter.  You can compute it from the shape of `samples`, which is a `compile time constant` and the number of MPI nodes, which is a runtime constant. 
Why are you passing it? Unless I'm missing something? 

`sparse_tol` and `sparse_maxiter` would be better not be compile time constants to avoid unnecessary recompilation. 
However, since we are jitting this block and therefore the cg inside I'm unsure if jax will correctly-pick up changes in sparse_tol. I suspect not.

Can you check that if you change the tolerance, something you can do without creating a new SR object and that the api supports, the jitted function will correctly pick up the new tolerance even without it being a static_argnum?

If not, it must be static.